podman_compose.py

```diff
@@ -2477,21 +2478,34 @@ def container_to_build_args(compose, cnt, args, path_exists):
         build_desc = {"context": build_desc}
     ctx = build_desc.get("context", ".")
     dockerfile = build_desc.get("dockerfile")
-    if dockerfile:
-        dockerfile = os.path.join(ctx, dockerfile)
-    else:
-        dockerfile_alts = [
-            "Containerfile",
-            "ContainerFile",
-            "containerfile",
-            "Dockerfile",
-            "DockerFile",
-            "dockerfile",
-        ]
-        for dockerfile in dockerfile_alts:
+    dockerfile_inline = build_desc.get("dockerfile_inline")
+    if dockerfile_inline is not None:
+        dockerfile_inline = str(dockerfile_inline)
+        # Error if both `dockerfile_inline` and `dockerfile` are set
+        if dockerfile and dockerfile_inline:
+            raise OSError("dockerfile_inline and dockerfile can't be used simultaneously")
+        # Create a temporary dockerfile
+        dockerfile = tempfile.NamedTemporaryFile(delete=False, suffix=".containerfile")
```

Comment:
Current code does not remove the temporary file, does it? I think this is primary deficiency of the current approach. Not sure how to best solve it right now.

One solution would be to introduce a list of cleanup functions that need to be called before podman-compose exits. Then pass this list to `container_to_build_args` which would append an appropriate os.remove call to that list. And then after `podman build` completes, remove the file in e.g. `finally`.